sql/rpl_rli.cc

```diff
@@ -2599,7 +2599,9 @@ void Relay_log_info::clear_relay_log_truncated() {
 }
 
 bool Relay_log_info::is_time_for_mts_checkpoint() {
-  if (is_parallel_exec() && opt_mts_checkpoint_period != 0) {
+  bool period_check = opt_mts_checkpoint_period != 0 &&
+                      !curr_group_seen_begin && !curr_group_seen_gtid;
```

Comment:
I seem to be seeing a ubsan runtime issue here:

```
../../sql/rpl_rli.cc:2603:50: runtime error: load of value 86, which is not a valid value for type 'bool'
    #0 0x6ac5253 in Relay_log_info::is_time_for_mts_checkpoint() sql/rpl_rli.cc:2603:50
    #1 0x6a66fa0 in exec_relay_log_event(THD*, Relay_log_info*, Rpl_applier_reader*) sql/rpl_slave.cc:4696:23
    #2 0x6a3b883 in handle_slave_sql sql/rpl_slave.cc:6826:9
    #3 0x81047da in pfs_spawn_thread(void*) storage/perfschema/pfs.cc:2836:3
    #4 0x7f63b9ff966d in start_thread (/usr/local/fbcode/platform007/lib/libpthread.so.0+0x766d)
    #5 0x7f63bb06fe2e in __GI___clone (/usr/local/fbcode/platform007/lib/libc.so.6+0x11ae2e)
```

I see curr_group_seen_begin initialized in the constructor, but not curr_group_seen_gtid. But it's not clear how we are skipping the initialization of these two variables in slave_start_workers(), but it looks like initializing curr_group_seen_gtid in the constructor removes the problem.